### ISSUE TITLE
TA DG Review

### ISSUE THREAD

**Issue by emer7** [AY1920S2-CS2103T-W13-4/main#177]
# Design
## 2.2
> Not updated

## 2.3
> ![image](https://user-images.githubusercontent.com/21377166/78469437-ea975a80-7753-11ea-9b69-e2b234e6cf8d.png)
> `StudentRemoveCommandParser` should end?

#176 # 2.4
> Not updated. Figure 7 I don't think reflect your app?

**Comment by emer7** [comment]
# Implementation
> Try to increase the variety of UML being used. Try using CD, OD, SD, or AD.

## 3.1.1.1, 3.1.2.1, 3.1.3.1, 3.1.4.1
> You have repetitive Sequence Diagram that is very similar to each other and does not really value add to your DG. Consider only to have 1 SD for all the similar one if possible
> You can consider including Class Diagram to illustrate all the classes that extends from Command (e.g. MinorSetCommand, etc.)

> ![image](https://user-images.githubusercontent.com/21377166/78470215-4533b500-775a-11ea-9f3a-adb1416a5f0c.png)
> Is that mistake?

## 3.2.1.1
> ![image](https://user-images.githubusercontent.com/21377166/78470244-77ddad80-775a-11ea-9243-4ba0ee16606b.png)
> Is that mistake?

## 3.3.1.1 , 3.3.3.1, 3.3.4.1, 3.4.1.1, 3.4.2.1, 3.4.3.1, 3.4.4.1
> ![image](https://user-images.githubusercontent.com/21377166/78470352-65b03f00-775b-11ea-9355-bff27c2484c8.png)
> Similar mistakes
> Blue: Consider extending the line so that it doesn't cause any misunderstanding
> Red: Is that mistake or intended? (does not apply to 3.3.4.1 and 3.4.4.1)

**Comment by emer7** [comment]
# Use Cases

> ![image](https://user-images.githubusercontent.com/21377166/78470715-f38d2980-775d-11ea-9f7c-ae2c693a8d73.png)
> Can add indentation if possible

> ![image](https://user-images.githubusercontent.com/21377166/78470734-1b7c8d00-775e-11ea-9373-7ca6d18108f7.png)
> Indentation too

> ![image](https://user-images.githubusercontent.com/21377166/78470764-639baf80-775e-11ea-80d4-e2b496073211.png)
> Numbering error (also the same for C.6.)
> Why for C.6, 4a resumes to step 3 while C.5 it ends immediately? Is this intended?

> ![image](https://user-images.githubusercontent.com/21377166/78470802-aa89a500-775e-11ea-8d40-d8269b57f700.png)
> Extension 2a does not make sense?

**Comment by emer7** [comment]
# NFRs
> Can be better and more focused to your app instead of generic / general app

# Glossary
> Too much unneeded term that does not relate to your app

**Comment by DANamic** [comment]
Hi @emer7, thank you so much for your thorough review and the feedback raised!

We've addressed them accordingly:
1. **Design** - We're made some major edits to that, as you're right - quite a bit was still referencing the old AB3, and we should be almost done with it by Sunday.
2. **Implementation** - Noted on the variety of diagrams required, so we are going to also try to include _Class Diagrams_ and maybe some small code snippets to help out too. We've also fixed some of the diagram errors you've pointed out!
3. **Use Cases** - For the indentation, we've standardised and fixed the indenting as best as possible, however for the `Use case ends.` or its variants on `Extensions`, we are not able to get them indented as you suggested, and have consulted the same with DGs from other groups. Nonetheless, we've fixed the inconsistent behaviour you've pointed out for C.5 and C.6,
4. **NFR** - We've edited this a little to make it less generic and more obvious that this is for NUS Undergraduate Students.
5. **Glossary** - We've removed any unused terms (that were originally expected), which should now make this section more succinct.

If there are no other comments, we will proceed this close this issue,

Thank you again, and stay safe!
